cvmfs/sync_union.cc

```diff
@@ -36,7 +36,9 @@ SharedPtr<SyncItem> SyncUnion::CreateSyncItem(
   PreprocessSyncItem(entry);
   if (entry_type == kItemFile) {
     entry->SetExternalData(mediator_->IsExternalData());
-    entry->SetCompressionAlgorithm(mediator_->GetCompressionAlgorithm());
+    if (!entry->IsValidGraft()) {
+      entry->SetCompressionAlgorithm(mediator_->GetCompressionAlgorithm());
```

Comment:
This might affect users who create graft files but do not use the `swissknife_graft` utility.  I'd prefer to make the `compression` key optionally so that graft files without the key would use `mediator_->GetCompressionAlgorithm()`.  Let's add a test for this case.

Reply:
Good point - I'll add this now.  Thank you!

Reply:
Added along with a test case